pkg/url/url_test.go

```diff
@@ -1637,3 +1638,388 @@ func TestGetContainerURL(t *testing.T) {
 		})
 	}
 }
+
+func TestListIngressURL(t *testing.T) {
+	// initialising the fakeclient
+	componentName := "testcomponent"
+	componentLabel := componentlabels.ComponentLabel
+	urlNameLabel := "odo.openshift.io/url-name"
+	port := intstr.IntOrString{
+		Type:   intstr.Int,
+		IntVal: int32(8080),
+	}
+	pushedIngress := []extensionsv1.Ingress{
```

Comment:
you can use the kclient fake ingress here

Reply:
updated